python/cugraph/community/leiden.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2019 - 2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.community import leiden_wrapper
+from cugraph.structure.graph import Graph
+
+
+def leiden(input_graph, max_iter=100, resolution=1.):
+    """
+    Compute the modularity optimizing partition of the input graph using the
+    Louvain heuristic
+
+    Parameters
+    ----------
+    input_graph : cugraph.Graph
+        cuGraph graph descriptor of type Graph
+
+        The adjacency list will be computed if not already present. The graph
+        should be undirected where an undirected edge is represented by a
+        directed edge in both direction.
+
+    max_iter : integer
+        This controls the maximum number of levels/iterations of the Louvain
+        algorithm. When specified the algorithm will terminate after no more
+        than the specified number of iterations. No error occurs when the
+        algorithm terminates early in this manner.
+
+    resolution: float/double, optional
+        Called gamma in the modularity formula, this changes the size
+        of the communities.  Defaults to 1.
+
+    Returns
+    -------
+    parts : cudf.DataFrame
+        GPU data frame of size V containing two columns the vertex id and the
+        partition id it is assigned to.
```

Comment:
partition, cluster, or communities? We should use consistent terminology in public API.